x/ecocredit/basket/msg_create.go

```diff
@@ -85,7 +85,7 @@ func (m MsgCreate) Route() string { return sdk.MsgTypeURL(&m) }
 func (m MsgCreate) Type() string { return sdk.MsgTypeURL(&m) }
 
 func validateDateCriteria(d *DateCriteria) error {
-	if d == nil {
+	if d == nil || d.String() == "" {
```

Comment:
> String() == ""

1. This is not an efficient way to check if it's an empty struct.
2. as discussed below  - we should use `nil` instead of setting empty date criteria.